### PR TITLE
Use bitmap to represent SFP errors

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**

### CODE REVIEW

File: sonic-xcvrd/xcvrd/xcvrd.py [hunk 1]
```diff
@@ -47,13 +47,14 @@
...
-# Convert the error code to string and store them in a set for convenience
-errors_block_eeprom_reading = set(str(error_code.value) for error_code in SFP_STATUS_ERR_ENUM)
+# SFP error code dictinary, new elements can be added if new errors need to be supported.
+SFP_STATUS_ERR_DICT = {
+    2: 'SFP_STATUS_ERR_I2C_STUCK',
+    4: 'SFP_STATUS_ERR_BAD_EEPROM',
+    8: 'SFP_STATUS_ERR_UNSUPPORTED_CABLE',
+    16: 'SFP_STATUS_ERR_HIGH_TEMP',
+    32: 'SFP_STATUS_ERR_BAD_CABLE'
+}
```
Comment (keboliu) [inline sonic-xcvrd/xcvrd/xcvrd.py]: maybe we still need to keep a set of "errors_block_eeprom_reading" as a subset of the whole errors, meaning not all errors will block the EEPROM reading, only on some certain error we will stop reading the EEPROM, but on others we will still continue to read.

Comment (Junchao-Mellanox) [reply sonic-xcvrd/xcvrd/xcvrd.py]: According to previous code implementation, all kinds of errors will be treated as "blocking eeprom read error". I suppose there will be no new errors in short term, could we keep it this way for now?

Comment (Junchao-Mellanox) [reply sonic-xcvrd/xcvrd/xcvrd.py]: @keboliu please let me know if you still have concern about this.

File: sonic-xcvrd/tests/test_xcvrd.py [hunk 2]
```diff
@@ -24,13 +24,15 @@
 test_path = os.path.dirname(os.path.abspath(__file__))
 modules_path = os.path.dirname(test_path)
 scripts_path = os.path.join(modules_path, "xcvrd")
-helper_file_path = os.path.join(scripts_path, "xcvrd_utilities"+"/y_cable_helper.py")
+#helper_file_path = os.path.join(scripts_path, "xcvrd_utilities"+"/y_cable_helper.py")
```
Comment (keboliu) [inline sonic-xcvrd/tests/test_xcvrd.py]: remove this line if not needed.

File: sonic-xcvrd/tests/test_xcvrd.py [hunk 3]
```diff
@@ -24,13 +24,15 @@
...
 sys.path.insert(0, modules_path)
 
 os.environ["XCVRD_UNIT_TESTING"] = "1"
-load_source('y_cable_helper', scripts_path + '/xcvrd_utilities/y_cable_helper.py')
-from y_cable_helper import *
+#load_source('y_cable_helper', scripts_path + '/xcvrd_utilities/y_cable_helper.py')
```
Comment (keboliu) [inline sonic-xcvrd/tests/test_xcvrd.py]: remove below two lines if not needed

**Review by keboliu — APPROVED** [review]

**Pull request closed without merging**